File: backend/app/dtos/model_availability_dto.py
from pydantic import BaseModel
class ModelAvailabilityDTO(BaseModel):


    model_download_name: str
    model_display_name: str
    model_description: str
    model_type: str

    downloaded: bool = False
    supported: bool = False
    usable_now: bool = False
    reason: str = ""

File: frontend/src/dtos/ModelAvailabilityDTO.ts
export type ModelAvailabilityDTO = {
	model_download_name: string;
	model_display_name: string;
	model_description: string;
	model_type: "transcription" | "summarization";
	downloaded: boolean;
	supported: boolean;
	usable_now: boolean;
	reason: string;
};
